src/containers/single-event/SingleEvent.tsx

```diff
@@ -0,0 +1,92 @@
+import { Spin } from 'antd';
+import React, { useEffect } from 'react';
+import { Helmet } from 'react-helmet';
+import { connect, useDispatch } from 'react-redux';
+import { useParams } from 'react-router-dom';
+import styled from 'styled-components';
+import EventDetails from '../../components/event-details/EventDetails';
+import { C4CState } from '../../store';
+import {
+  AsyncRequest,
+  asyncRequestIsComplete,
+  asyncRequestIsFailed,
+  asyncRequestIsNotStarted,
+} from '../../utils/asyncRequest';
+import { getUpcomingEvents } from '../upcoming-events/ducks/thunks';
+import { EventProps, EventsReducerState } from '../upcoming-events/ducks/types';
+
+const ContentContainer = styled.div`
+  padding: 24px;
+  margin: auto;
+  max-width: 1200px;
+  width: 80%;
+`;
+
+const CenteredContainer = styled.div`
+  display: flex;
+  justify-content: center;
+`;
+
+interface UpcomingEventsProps {
+  readonly events: EventsReducerState['upcomingEvents'];
+}
+
+interface SingleEventParams {
+  id: string;
+}
+
+const SingleEvent: React.FC<UpcomingEventsProps> = ({ events }) => {
+  const dispatch = useDispatch();
+  useEffect(() => {
+    if (asyncRequestIsNotStarted(events)) {
```

Comment:
What if the request is failed? Should we retry?

Reply:
How do we avoid a retry infinite loop?

Reply:
When you dispatch, it goes into the loading state automatically, so the state will no longer be failed.